src/main/java/org/commcare/formplayer/api/json/PromptToJson.java

```diff
@@ -201,8 +202,10 @@ private static JSONArray parseSelect(FormEntryPrompt prompt) {
 
     //TODO WSP: What the fuck is drew doing XFormPlayer parse_style_info
     // https://github.com/dimagi/touchforms/blob/master/touchforms/backend/xformplayer.py#L400
-    private static JSONObject parseStyle(FormEntryPrompt prompt) {
+    private static JSONObject parseStyle(FormEntryModel model) {
+        FormEntryCaption prompt = model.getCaptionPrompt();
```

Comment:
This is a bit of a bigger change than it seems, since it makes this method signature dependent on a mutable state in `FormEntryModel` which has an internal walker, rather than an immutable object.

I would push this call upwards, into the `parseQuestionType` method, and generalize this method's signature to receive a `FormEntryCaption` (which is a supertype of a `FormEntryPrompt`) but has the same stateless semantics.

Reply:
Sounds good, updated in e94281b0421e975bc38f5bc53ac271e9269ece6d - let me know if that's not what you meant.